fossunited/api/profile.py

```diff
@@ -78,17 +77,23 @@ def update_profile(fields_dict):
             "mastodon": fields_dict.get("mastodon"),
         }
 
-        frappe.db.set_value(USER_PROFILE, user_doc.name, updated_fields)
+        profile = frappe.get_doc(USER_PROFILE, user_doc.name)
+        for field, value in updated_fields.items():
+            if hasattr(profile, field):
+                setattr(profile, field, value)
+        profile.save(ignore_permissions=True)
 
         user_updates = {}
         if fields_dict.get("full_name") != user_doc.full_name:
-            user_updates["full_name"] = fields_dict.get("full_name")
+            user_updates["first_name"] = fields_dict.get("full_name").split(" ")[0]
```

Comment:
it's not clear why we only update `first_name` when `full_name` is provided

Reply:
fixed this to update first and last name